custom-vu/src/main/kotlin/jces1209/vu/page/AttachScreenShot.kt

```diff
@@ -0,0 +1,33 @@
+package jces1209.vu.page
+
+import jces1209.vu.wait
+import org.openqa.selenium.By
+import org.openqa.selenium.OutputType
+import org.openqa.selenium.TakesScreenshot
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions
+
+abstract class AttachScreenShot(
+    protected val driver: WebDriver
+) {
+    protected var countBefore: Int = 0
+    abstract val screenShotXPath: String
```

Comment:
1. It looks redundant in the parent class because it is not being used in the parent but adds some restrictions for children
2. screenShotXPath is being used with By.xpath each time - you can just make this one
```suggestion
    protected var countBefore: Int = 0
    abstract val screenShotLocator: By
```